src/k8s-engine/lima.ts

```diff
@@ -901,7 +928,7 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
     explanations.push(VDE_DIR);
   }
 
-  protected async createLimaSudoersFile(commands: Array<string>, explanations: Array<string>, randomTag: string): Promise<void> {
+  protected async createLimaSudoersFile(this: Readonly<this> & this, commands: Array<string>, explanations: Array<string>, randomTag: string): Promise<void> {
```

Comment:
If this is an intersection type, is this just the list of fields that the `Readonly` doesn't affect? Whether yes or no, what's going on here?

Reply:
Here, `this` (within the function body) is the combination of 1) the public members of `LimaBackend`, but read-only; and 2) the non-public members of `LimaBackend`.  Unfortunately, this is necessary because `Readonly<T>` is incorrect in that it only contains the _public_ members of `T`, not the `protected` &c. ones.  As we do need access to protected members (specifically, `this.lima`), we need to do the intersection even those this means we lose some of the protection from typing.